### PR TITLE
Add automatic __init__

### PR DESCRIPTION
**Pull request by PythonFZ**
# TODO
- [x] Some sort of autocompletion / update the docs
- [x] fix merge conflicts

# API

Allow for

```py
class HelloWorld(Node):
   param1 = zn.params()
   result = zn.outs()

   def run(self):
      self.result = self.param

HelloWorld(param=25) # without defining the __init__
```

the created init is then `__init__(self, param1=None)`

### CODE REVIEW

File: zntrack/core/base.py [hunk 1]
```diff
@@ -48,6 +49,21 @@ def __init__(self, **kwargs):
...
+        if cls.__dict__.get("__init__") is not None:
+            return cls
+
+        # attach an automatically generated __init__ if None is provided
+        zn_option_fiels = []
+        for name, item in cls.__dict__.items():
+            if isinstance(item, params):
+                zn_option_fiels.append(name)
+
+        setattr(cls, "__init__", get_auto_init(fields=zn_option_fiels))
```
Comment (PythonFZ) [inline zntrack/core/base.py:65]: new init is created

Comment (JR-1991) [reply zntrack/core/base.py]: Committed the "__signature__" addition to this pull request. ZnTrack parameters are now displayed in the signature. Tested with Jupyter Lab and the code given above.

**Review by JR-1991 — APPROVED** [review]
Meets the requirements now :-)

File: zntrack/core/base.py (hunk 1 above)
Comment (PythonFZ) [reply zntrack/core/base.py]: Awesome - the final addition for the next release.

**Pull request merged**